src/lib/components/ConfirmTransactions.js

```diff
@@ -12,78 +14,189 @@ ${json}
-                  <span className="text-secondary">Deposit: </span>
+                  <h4>Transaction #{i + 1}</h4>
+                </div>
+                <div>
+                  <span className="text-secondary">Contract ID: </span>
                   <span className="font-monospace">
-                    {displayNear(transaction.deposit)}
+                    {transaction.contractName}
                   </span>
                 </div>
-              )}
-              <div>
-                <span className="text-secondary">Gas: </span>
-                <span className="font-monospace">
-                  {displayGas(transaction.gas)}
-                </span>
+                <div>
+                  <span className="text-secondary">Method name: </span>
+                  <span className="font-monospace">{transaction.methodName}</span>
+                </div>
+                {transaction.deposit && transaction.deposit.gt(0) && (
+                  <div>
+                    <span className="text-secondary">Deposit: </span>
+                    <span className="font-monospace">
+                      {displayNear(transaction.deposit)}
+                    </span>
+                  </div>
+                )}
+                <div>
+                  <span className="text-secondary">Gas: </span>
+                  <span className="font-monospace">
+                    {displayGas(transaction.gas)}
+                  </span>
+                </div>
+                <Markdown text={jsonMarkdown(transaction.args)} />
               </div>
-              <Markdown text={jsonMarkdown(transaction.args)} />
+            ))}
+          {transactions.length == 1 ?
+            <div class="form-check form-switch">
+              <input class="form-check-input" type="checkbox" id="dontaskagaincheckbox"
+                checked={dontAskAgainChecked}
+                onChange={() => dontAskAgainCheckboxChange()}
+              />
+              <label class="form-check-label" for="dontaskagaincheckbox">
+                Don't ask again for sending similar transactions by{" "}
+                <span className="font-monospace">{widgetSrc}</span>
+              </label>
             </div>
-          ))}
-      </Modal.Body>
-      <Modal.Footer>
-        <button
-          className="btn btn-success"
-          disabled={loading}
-          onClick={(e) => {
-            e.preventDefault();
-            setLoading(true);
-            near.sendTransactions(transactions).then(() => {
+            :
+            <></>
+          }
+        </Modal.Body>
+        <Modal.Footer>
+          <button
+            className="btn btn-success"
+            disabled={loading}
+            onClick={async (e) => {
+              e.preventDefault();
+              setLoading(true);
+              if (dontAskAgainChecked) {
+                const contractId = transactions[0].contractName;
+                const methodName = transactions[0].methodName;
```

Comment:
As you see on line [139](https://github.com/NearSocial/VM/pull/156/files/22e8585d60da1b08eaa1889489bb48fa27f18341#diff-2816d3f924c1dbc749b117f468f07ebf91ea0e3fc8abe8dfa1bacd8e10162ea3R139) there is a check if the number of transactions is `1`. Otherwise, it will not offer the option to not ask again.